frontend/pages/sign-in/code.tsx

```diff
@@ -0,0 +1,192 @@
+    if (!isUserLoading && !!user) {
+      if (user?.role === UserRoles.Light) {
+        // If the user don't have an account go to the 'choose account type' page
+        replace(Paths.AccountType);
+      } else {
+        // If the user has an account go to the dashboard
+        replace(Paths.Dashboard);
+      }
+    }
+  }, [isUserLoading, replace, user]);
+
+  // Create an array of numbers from 0 to the code's length
+  const code = Array.from({ length: CODE_LENGTH }, (_, i) => i);
+
+  useEffect(() => {
+    // Set focus at the first input box
+    setFocus('otp_attempt.0');
+  }, [setFocus]);
+
+  const handleChangeCode = (e: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.currentTarget;
+    const currentInputIndex = Number(name.split('.')[1]);
+    let nextInput = currentInputIndex + 1;
+    if (!value || currentInputIndex === 4) return;
+    setFocus(`otp_attempt.${nextInput}`);
+  };
+
+  const handleSignIn = useCallback(
+    (data: SignIn) => {
+      signIn.mutate(data, {
+        onSuccess: () => {
+          replace(Paths.Dashboard);
+        },
+      });
+    },
+    [signIn, replace]
+  );
+
+  const handlePasteCode = (ev: ClipboardEvent<HTMLInputElement>) => {
+    const text = ev.clipboardData.getData('text');
+    text
+      .split('')
+      .slice(0, CODE_LENGTH)
+      .forEach((char, index) => setValue(`otp_attempt.${index}`, char));
+  };
+
+  const onSubmit: SubmitHandler<SignInCodeForm> = (code) => {
+    const otp_attempt = Object.values(code.otp_attempt).join('');
+    handleSignIn({ ...sessionData, otp_attempt });
+  };
+
+  return (
+    <div className="flex flex-col items-center justify-center w-full h-screen max-w-xl m-auto">
+      <div>
+        <div className="mb-6 text-center">
+          <p className="mb-6 font-semibold">HeCo invest</p>
+          <h1 className="font-serif text-3xl font-semibold text-green-dark">
+            <FormattedMessage defaultMessage="Insert your code" id="Nn0ig7" />
+          </h1>
+        </div>
+
+        <form onSubmit={handleSubmit(onSubmit)} noValidate>
+          <div className="w-full max-w-md text-center">
+            <fieldset>
+              <legend className="mb-6 text-gray-800">
+                <FormattedMessage
+                  defaultMessage="To help keep your account secure, we sent a security code to your email. "
+                  id="/vHsVH"
+                />
+              </legend>
+              <div className="flex justify-center gap-2.5">
+                {code.map((digit) => (
+                  <Input
+                    key={digit}
+                    type="text"
+                    name={`otp_attempt.${digit}`}
+                    registerOptions={{
+                      onChange: handleChangeCode,
+                      maxLength: 1,
+                      required: true,
+                    }}
+                    aria-describedby="code-error"
```

Comment:
The `<fieldset />`'s `legend` is not enough to make these inputs accessible. I think that if you add this prop, it's way clearer when using a screen reader:
```suggestion
                    aria-describedby="code-error"
                    aria-label={`Two factor authentication code, digit ${digit + 1}`}
```